colossalai/fx/passes/algorithms/ckpt_solver_chen.py

```diff
@@ -1,45 +1,71 @@
+from typing import Set, Tuple
 import torch
 from torch.fx import GraphModule
+import math
 
 __all__ = ['chen_greedy', 'chen_sqrtn']
 
 
-def chen_greedy(gm: GraphModule, B: int):
+def chen_greedy(gm: GraphModule) -> GraphModule:
     """
     This is the simple implementation of Algorithm 3 in https://arxiv.org/abs/1604.06174.
+    Note that this algorithm targets at memory optimization only, using techniques in appendix A.
 
     Usage:
-        B = 5 * 1024 * 1024 * 1024  # An approximate memory budget of 5GB
         model = resnet18()
         input_sample = torch.rand(4, 3, 224, 224)
         gm = symbolic_trace(model)
         MetaInfoProp(gm).run(input_sample)
-        gm = chen_greedy(gm, B)
+        gm = chen_greedy(gm)
 
     Args:
         gm (GraphModule): The module to add checkpoints
-        B (int): The approximate memory budget for this module.
     """
+
+    def grid_search(num_grids: int = 6) -> Set:
+        """
+        Search ckpt strategy with B = 0, then run the allocation algorithm again with B = √xy.
+        Grid search over [√2/2 B, √2 B] for ckpt_opt over num_grids as in appendix A.
+        """
+        _, B_approx = run_chen_greedy(0)
+        B_min, B_max = math.floor(B_approx / math.sqrt(2)), math.ceil(B_approx * math.sqrt(2))
+        B_opt = math.inf
+        for B in range(B_min, B_max, (B_max - B_min) // num_grids):
+            ckpt, B_approx = run_chen_greedy(B)
+            if B_approx < B_opt:
+                B_opt = B_approx
```

Comment:
do not start a variable with capital letter.

Reply:
I was trying to follow the notations in the paper.

Reply:
I understand, but it will  be good to stick with the python code standard.

Reply:
I see

Reply:
Agree, just change the variables names